### PR TITLE
[Set] Handling possible custom set list

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/5891

### CODE REVIEW

File: bin/rector.php [hunk 1]
```diff
@@ -48,6 +49,16 @@
...
 } catch (SetNotFoundException $setNotFoundException) {
+    // possible custom set list
+    $match = Strings::match($setNotFoundException->getMessage(), '#Set (".*") was not found#');
+    if ($match) {
+        $path = realpath(trim($match[1], '"'));
+        if ($path) {
+            // eg : /Users/samsonasik/www/laminas-mvc-skeleton/vendor/laminas/laminas-servicemanager-migration/config/rector/set/laminas-servicemanager-40.php
+            echo $path;die;
+        }
+    }
```
Comment (samsonasik) [inline bin/rector.php:60]: Hi @TomasVotruba , this is where the path is catched when we define custom `SetList` class, do you have idea to make it work to read `$path` as load config from it ?

ref https://github.com/rectorphp/rector/issues/5891
ref https://github.com/laminas/laminas-servicemanager-migration/pull/2#issuecomment-801530436

Comment (samsonasik) [reply bin/rector.php]: @TomasVotruba it works 🎉 , I updated to try re-create container with path found https://github.com/rectorphp/rector/pull/5894/commits/d96afeca8d4d8570ee341eddef88288649b6194b

File: bin/rector.php [hunk 2]
```diff
@@ -47,9 +50,26 @@
...
+                $invalidSetReporter = new InvalidSetReporter();
+                $invalidSetReporter->report($setNotFoundException);
+                exit(ShellCode::ERROR);
+            }
+        }
+    } else {
+        $invalidSetReporter = new InvalidSetReporter();
+        $invalidSetReporter->report($setNotFoundException);
+        exit(ShellCode::ERROR);
+    }
```
Comment (TomasVotruba) [inline bin/rector.php:72]: This should not require a container to boot. Purpose of this is to keep thin bin file.

Better approach might something `MissingRectorRulesReporter`, only focused on sets:
https://github.com/rectorphp/rector/blob/main/src/Reporting/MissingRectorRulesReporter.php

Comment (samsonasik) [reply bin/rector.php:72]: I don't understand, as at this point, the custom `Laminas\ServiceManager\Migration\Rector\Set\ValueObject\SetList` is never catched as not registered at 

https://github.com/rectorphp/rector/blob/c815334bed781c555f08dff82ce7861d758e9e7a/packages/Set/RectorSetProvider.php#L30

The get by error message is got a full path of set list constant value.

The `MissingRectorRulesReporter` seems only for reporting, not make the custom `SetList` works.

How to make it work if not re-create the `$container` ? is

Comment (samsonasik) [reply bin/rector.php:72]: Tried add both `parameters` and `services`. The `services` part not works, only the parameters part.

**Comment by samsonasik** [comment]
I am closing it as the solution can use `$containerConfigurator->import()` like commented at https://github.com/rectorphp/rector/issues/5891#issuecomment-803291383

**Pull request closed without merging**